libs/dal/src/repositories/job/job.repository.ts

```diff
@@ -150,21 +153,16 @@ export class JobRepository extends BaseRepository<EnforceEnvironmentQuery, JobEn
     selectNotification?: ProjectionType<NotificationEntity>;
     selectSubscriber?: ProjectionType<SubscriberEntity>;
     selectEnvironment?: ProjectionType<EnvironmentEntity>;
-  }): Promise<
-    JobEntity & {
-      template: NotificationTemplateEntity;
-      notification: NotificationEntity;
-      subscriber: SubscriberEntity;
-      environment: EnvironmentEntity;
-    }
-  > {
-    return this.MongooseModel.findOne(query, select)
+  }) {
+    const job = this.MongooseModel.findOne(query, select)
       .populate('template', selectTemplate)
       .populate('notification', selectNotification)
       .populate('subscriber', selectSubscriber)
       .populate('environment', selectEnvironment)
       .lean()
       .exec();
+
+    return job as unknown as JobEntityPopulated;
```

Comment:
here we don't do mapping, I don't know why, but I see its failing when calling `this.mapEntity(job)` and returns `{}` empty object

Reply:
Try to replace `this.MongooseModel` for `this._model` in line 157 and see if that's the reason.

Reply:
@p-fernandez that is not the reason, base repository sets the `this._model` in the constructor, the same as it does for `this.MongooseModel`. The problem might be the `JSON.stringify(entity)` call that does the `this.mapEntity`